### PR TITLE
Tests/pkg_fatfs_vfs: make test fail correctly for samr21-xpro

### PR DESCRIPTION
**Pull request by danpetry**
### Contribution description

The tests/pkg_fatfs_vfs previously hung, rather than failing, which it should do if there's no SD card present.

The reason for the hanging was that the timeouts in the sdcard_spi module were very long. Once this was reduced, the test script would fail, but by throwing an exception rather than running fully and giving an error code. This is different from the behaviour of other tests and IMO not the best way to do it: an exception should really indicate a failure/error of the test script, not the program under test.

### Testing procedure

`make test` with a SAMR21-xpro.

### CODE REVIEW

**Comment by miri64** [comment]
@MichelRottleuthner can you please have a look. I'm not sure if changing some timings is the right appoach here.

**Comment by danpetry** [comment]
yeah, this solution was something more of a quick fix that doesn't break anything, with more extended (proper) discussion/rework to be addressed in the linked issue https://github.com/RIOT-OS/RIOT/issues/11388. Would be interested to see what other opinions people have on it.

**Comment by miri64** [comment]
> yeah, this solution was something more of a quick fix that doesn't break anything, with more extended (proper) discussion/rework to be addressed in the linked issue #11388. Would be interested to see what other opinions people have on it.

For now:  would increasing the timeout on the test's side help? I got news from @MichelRottleuthner that he is currently short on time, so I don't know if he is able to review this week.

File: tests/pkg_fatfs_vfs/tests/01-run.py [hunk 1]
```diff
@@ -9,26 +9,15 @@
...
 
     child.expect(u"Tests for FatFs over VFS - test results will be printed in "
                  "the format test_name:result\r\n")
 
-    while True:
-        res = child.expect([u"[^\n]*:\[OK\]\r\n",
-                            u"Test end.\r\n",
-                            u".[^\n]*:\[FAILED\]\r\n",
-                            u".*\r\n"])
-        if res > 1:
```
Comment (miri64) [inline tests/pkg_fatfs_vfs/tests/01-run.py]: Also why is this test removed? Seems quite important to me to check error states.

**Comment by danpetry** [comment]
Ok thanks for explaining, yes I didn't consider the loop/multiple test output you described above. See what you're saying as well re the exception and that it's generally in line with other test philosophies, for better or worse.

The problem I was trying to solve here was that this test gives a false negative result during the release specs testing (specifically test 02). It will fail for boards where supporting hardware (an SD card) doesn't exist. I had a look at a couple of the driver tests and saw that these just expect `[FAILURE]` or the like without the test script returning any failure behaviour/information/etc to the shell. So I tried to emulate this behaviour. However, seems like this behaviour should be fixed rather than emulated.

@cladmi says he's familiar with the issue with this test and says that the proper solution is to define a makefile variable that indicates hardware-based dependencies on tests. Then the test would be excluded from being automatically run.

In summary: it doesn't look like there's a satisfactory "quick fix" for the problem. So, better to document that the tester will need to know that this test gives a false negative.

The separate problem of the long timeouts then should also be addressed "properly" and the issue above is for that.

therefore closing the PR.
